### ISSUE TITLE
Javadoc?

### ISSUE THREAD

**Issue by davisjam** [mauricioaniche/repodriller#70]
Should this project have Javadoc e.g. to describe the classes and their methods?

For example, I was surprised to see that the PersistenceMechanism(s) given to RepositoryMining.process will be auto-closed, and could only find this out from the source.

**Comment by mauricioaniche** [comment]
Yes, definitely! Feel free to add any documentation. I'll make sure to write them whenever I have some change to do.
